src/pip/_internal/resolution/resolvelib/resolver.py

```diff
@@ -87,7 +87,8 @@ def resolve(
         if "PIP_RESOLVER_DEBUG" in os.environ:
             reporter: BaseReporter[Requirement, Candidate, str] = PipDebuggingReporter()
         else:
-            reporter = PipReporter()
+            reporter = PipReporter(constraints=provider._constraints)
```

Comment:
Rather than accessing a private attribute add a property to `PipProvider` and then use `provider.constraints`:

```
    @property
    def constraints(self) -> dict[str, Constraint]:
        return self._constraints
```